docs/developer/jobs.md

```diff
@@ -0,0 +1,90 @@
+# Jobs
+
+This document intents to provide an overview over our jobs running via Prow, GitHub actions and Google Cloud Build.
+
+#### Builds and Tests
+
+Prow Presubmits:
+* [pull-cluster-api-build-main] ./scripts/ci-build.sh
+* \[opt\] \[always-run:false\] [pull-cluster-api-make-main] ./scripts/ci-make.sh
+* \[opt\] [pull-cluster-api-apidiff-main] ./scripts/ci-apidiff.sh
+* [pull-cluster-api-verify] ./scripts/ci-verify.sh
+* [pull-cluster-api-test-main] ./scripts/ci-test.sh
+* [pull-cluster-api-e2e-main] ./scripts/ci-e2e.sh
+    * GINKGO_FOCUS: `[PR-Blocking]`
+* \[opt\] [pull-cluster-api-e2e-ipv6-main] ./scripts/ci-e2e.sh
+    * GINKGO_FOCUS: `[PR-Blocking]`, IP_FAMILY: `IPv6`
+* \[opt\] \[always-run:false\] [pull-cluster-api-e2e-full-main] ./scripts/ci-e2e.sh
+    * GINKGO_SKIP: `[PR-Blocking] [Conformance] [K8s-Upgrade]` (i.e. "no tags")
+* \[opt\] \[always-run:false\] [pull-cluster-api-e2e-workload-upgrade-1-21-latest-main] ./scripts/ci-e2e.sh FROM: stable-1.21 TO: ci/latest-1.22
+    * GINKGO_FOCUS: `[K8s-Upgrade]`
+
+Prow Periodics:
+* [periodic-cluster-api-verify-book-links-main] make verify-book-links
+* [periodic-cluster-api-test-main] ./scripts/ci-test.sh
+* [periodic-cluster-api-e2e-main] ./scripts/ci-e2e.sh
+    * GINKGO_SKIP: `"[Conformance] [K8s-Upgrade]`
+* [periodic-cluster-api-e2e-workload-upgrade-1-18-1-19-main] ./scripts/ci-e2e.sh FROM: stable-1.18 TO: stable-1.19
+    * GINKGO_FOCUS: `[K8s-Upgrade]`
+* [periodic-cluster-api-e2e-workload-upgrade-1-19-1-20-main] ./scripts/ci-e2e.sh FROM: stable-1.19 TO: stable-1.20
+    * GINKGO_FOCUS: `[K8s-Upgrade]`
+* [periodic-cluster-api-e2e-workload-upgrade-1-20-1-21-main] ./scripts/ci-e2e.sh FROM: stable-1.20 TO: stable-1.21
+    * GINKGO_FOCUS: `[K8s-Upgrade]`
+* [periodic-cluster-api-e2e-workload-upgrade-1-21-latest-main] ./scripts/ci-e2e.sh FROM: stable-1.21 TO: ci/latest-1.22
+    * GINKGO_FOCUS: `[K8s-Upgrade]`
+* [cluster-api-push-images-nightly] Google Cloud Build: make release-staging-nightly, make -C test/infrastructure/docker release-staging-nightly
+
+Prow Postsubmits:
+* [post-cluster-api-push-images] Google Cloud Build: make release-staging, make -C test/infrastructure/docker release-staging
+
+Github (Presubmit) Workflows:
+* golangci-lint: golangci/golangci-lint-action@v2
+* verify: kubernetes-sigs/kubebuilder-release-tools@v0.1 verifier
+
+Github (On Release) Workflows:
+* Update Homebrew Formula On Release: dawidd6/action-homebrew-bump-formula@v3 clusterctl
+
+#### e2e tests
+
+* no tags:
+    * capi-e2e.When testing KCP upgrade in a single control plane cluster Should successfully upgrade Kubernetes, DNS, kube-proxy, and etcd
```

Comment:
how do we make sure this doc stays up to date? Seems like this level of detail will require quite a bit of maintenance if we don't want it to become outdated

Reply:
What I'm suggesting is that a higher level overview of what each job does and why it's required (or not required) might be a better long term effort:value investment

Reply:
Definitely very hard to keep up to date. I can drop this part. The idea was to provide an overview which e2e tests are executed in which jobs. This can mostly also be achieved by clicking at the test grid links above (apart from the upgrade tests which show the conformance test results instead of the e2e test results)

Reply:
> This can mostly also be achieved by clicking at the test grid links above (apart from the upgrade tests which show the conformance test results instead of the e2e test results)


Maybe an easy win is to document how to do that instead? "To see which e2e tests are executed in which jobs, ..."

Reply:
Done